ReactSkia/components/RSkComponent.cpp

```diff
@@ -80,18 +82,29 @@ RnsShell::LayerInvalidateMask RSkComponent::updateProps(const ShadowView &newSha
       layer_->shadowRadius = newviewProps.shadowRadius;
       /*TODO : To be tested and confirm updateMask need for this Prop*/
       updateMask =static_cast<RnsShell::LayerInvalidateMask>(updateMask | RnsShell::LayerInvalidateAll);
+      creatShadowFilter=true;
    }
   //shadowoffset
    if ((forceUpdate) || (oldviewProps.shadowOffset != newviewProps.shadowOffset)) {
       layer_->shadowOffset = RSkSkSizeFromSize(newviewProps.shadowOffset);
       /*TODO : To be tested and confirm updateMask need for this Prop*/
       updateMask =static_cast<RnsShell::LayerInvalidateMask>(updateMask | RnsShell::LayerInvalidateAll);
+      creatShadowFilter=true;
    }
   //shadowcolor
    if ((forceUpdate) || (oldviewProps.shadowColor != newviewProps.shadowColor)) {
       layer_->shadowColor = RSkColorFromSharedColor(newviewProps.shadowColor,SK_ColorBLACK);
       /*TODO : To be tested and confirm updateMask need for this Prop*/
       updateMask =static_cast<RnsShell::LayerInvalidateMask>(updateMask | RnsShell::LayerInvalidateAll);
+      creatShadowFilter=true;
+   }
+   if((layer_->shadowFilter==NULL) || creatShadowFilter) {
+       layer_->shadowFilter= SkImageFilters::DropShadowOnly(
+                              layer_->shadowOffset.width(), layer_->shadowOffset.height(),
+                              layer_->shadowRadius, layer_->shadowRadius,
+                              layer_->shadowColor, nullptr);
+   } else {
```

Comment:
is this else condition is valid now ?
because if it does not require update , the shadowFilter will be reset

Reply:
yes not require else condition.